### ISSUE TITLE
Promtail label __meta_docker_container_log_stream missing in docker_sd_config

### ISSUE THREAD

**Issue by Mario-Hofstaetter** [grafana/loki#5940]
**Describe the bug**

When [using `docker_sd_config`](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#docker_sd_config) there should be a metadata label 

`__meta_docker_container_log_stream: the log stream type stdout or stderr`

However, it does not seem to exist. It is also not mentioned in the prometheus documentation:
https://prometheus.io/docs/prometheus/latest/configuration/configuration/#docker_sd_config

It was also not removed from docs or sourcecode, it seems it never existed?
https://github.com/prometheus/prometheus/blob/01c55822169a9123367e333dc407edd8a263f75d/discovery/moby/docker.go#L38

## Impact on `loki-canary`

Because of this, it does not seem possible to run `loki-canary` and `promtail` as docker-containers when using `docker_sd_configs`.
[canary requires two stream selectors](https://grafana.com/docs/loki/latest/operations/loki-canary/#configuration), set via `-labelname`/`-labelvalue` and `-streamname` / `-streamvalue`

I tried to use available labels (`job` and `container_name`) instead, but canary is going crazy because it does not expect other log-messages it writes on `stderr` , but only the `1650132968082754000 ppppppppppppppppppppppppppppppppppppppppppp` on stdout.

It starts looping and logging these:

```
received invalid entry: received invalid entry: received invalid entry: received invalid entry: received invalid entry: received invalid entry: received an unexpected entry with ts 1650132864614161600
```

**To Reproduce**

- Run promtail using a job with `docker_sd_configs`
- Examine the labels available during relable pahse (targets web interface)
- `__meta_docker_container_log_stream` is missing

**Expected behavior**

`__meta_docker_container_log_stream` should exists as documented.

**Environment:**
 - Docker image `grafana/promtail:2.5.0`
 - Docker on windows 10

**Screenshots, Promtail config, or terminal output**

<details>
<summary> promtail-config.yaml (CLICK ME)</summary>

```yaml
server:
  http_listen_address: 0.0.0.0
  http_listen_port: 9080

clients:
  - url: http://loki:3100/loki/api/v1/push
    tls_config:
      insecure_skip_verify: true

scrape_configs:
  - job_name: docker-container-logs
    docker_sd_configs:
      - host: unix:///var/run/docker.sock
        refresh_interval: 30s

    relabel_configs:
      - source_labels: ['__meta_docker_container_name']
        regex: '/(.*)'
        target_label: 'container_name'

      - source_labels: ['__meta_docker_container_id']
        regex: '(.*)'
        target_label: 'container_id'

      - source_labels: ['__meta_docker_container_log_stream']
        regex: '(.*)'
        target_label: 'container_log_stream'
```

</details>

<details>
<summary> docker-compose.yaml (CLICK ME)</summary>

```yaml
version: "2.4"

services:
  promtail:
    command: "-config.file=/etc/loki/promtail-config.yaml"
    image: grafana/promtail:${TAG_PROMTAIL:-2.5.0}
    ports:
      - 9080:9080
    volumes:
      - ./promtail-config.yaml:/etc/loki/promtail-config.yaml:ro
      - /var/run/docker.sock:/var/run/docker.sock

```

</details>

![grafik](https://user-images.githubusercontent.com/33002073/163686230-c4d88a30-317d-4512-b93f-eb013a2ac9a0.png)

**Comment by cstyan** [comment]
Hmmm, this is likely something that would need to be fixed in upstream prometheus' docker SD since we just reuse their SD libraries.

**Comment by Mario-Hofstaetter** [comment]
> Hmmm, this is likely something that would need to be fixed in upstream prometheus' docker SD since we just reuse their SD libraries.

Yes, curious to know however how it got into the docs in the first place if it never was available?
Maybe an upstream PR that never got merged?

**Comment by jeschkies** [comment]
I believe Promtail 2.5.0 is missing https://github.com/grafana/loki/pull/5890. This issue seems to be a duplicate of https://github.com/grafana/loki/issues/5882.

**Comment by Mario-Hofstaetter** [comment]
> I believe Promtail 2.5.0 is missing #5890. This issue seems to be a duplicate of #5882.

I can confirm the issue has been fixed by #5890 , tested with `grafana/promtail:main` , `version=main-f62b4ae, branch=main, revision=f62b4ae19`
